src/main/kotlin/no/nav/syfo/service/SenderFacade.kt

```diff
@@ -76,29 +83,45 @@ class SenderFacade(
         content: String,
         url: URL? = null,
         varselHendelse: ArbeidstakerHendelse,
-        meldingType: BrukernotifikasjonKafkaProducer.MeldingType? = BrukernotifikasjonKafkaProducer.MeldingType.BESKJED,
+        varseltype: InternalBrukernotifikasjonType,
         eksternVarsling: Boolean = true,
         smsContent: String? = null,
     ) {
-        var isSendingSucceed = true
         try {
-            brukernotifikasjonerService.sendVarsel(uuid, mottakerFnr, content, url, meldingType, eksternVarsling, smsContent)
+            when (varseltype) {
+                InternalBrukernotifikasjonType.BESKJED -> brukernotifikasjonerService.sendBeskjed(
+                    uuid = uuid,
+                    mottakerFnr = mottakerFnr,
+                    content = content,
+                    url = url,
+                    eksternVarsling = eksternVarsling
+                )
+
+                InternalBrukernotifikasjonType.OPPGAVE -> brukernotifikasjonerService.sendOppgave(
+                    uuid = uuid,
+                    mottakerFnr = mottakerFnr,
+                    content = content,
+                    url = url,
+                    smsContent = smsContent
+                )
+
+                InternalBrukernotifikasjonType.DONE -> brukernotifikasjonerService.ferdigstillVarsel(uuid = uuid)
+            }
```

Comment:
Hva vinner vi egentlig med å flytte denne logikken hit?

Reply:
Nå code reviewer du godt her :P 

Jeg ville egentlig gjøre mer refaktorering enn jeg gjorde i den oppgaven her, og ville helst fjerne hele senderfacade og heller ha "domenestrukturerte" mapper. Men så ble det for mye scope creep.

Enig i at å bare flytte den ikke gir så mye verdi, kan godt flytte det inn i servicen om du vil?

Reply:
Ja, vi kan sikkert ta en diskusjon om det senere og bli enig om hvordan vi vil ha det. Utgangspunktet med facaden var at den skulle håndtere lagring av utsendte og feilede varsler i databasen, men at den ellers skulle være ganske tynn. Dersom det blir aktuelt med kanal-spesifikke kafka-topics i fremtiden så er det muligens også en fordel å beholde det i servicen.

Reply:
Ellers er jeg helt for mer domenestrukturerte mapper i stedet for consumer- og producer-mapper for eksempel.

Reply:
Ok. Flytter inn :)